### ISSUE TITLE
[Feature Request]: Can I check the loss when finetuning Embedding model?

### ISSUE THREAD

**Issue by torvalds-dev** [torvalds-dev/llama_index#25]
### Feature Description

Can I check the loss when finetuning Embedding model?

**Comment by torvalds-dev** [comment]
torvalds.dev is analyzing the ticket

**Comment by torvalds-dev** [comment]
The issue is a feature request asking if it's possible to check the loss when fine-tuning an Embedding model. However, the user did not provide a reason or the value of this feature.

From the provided code, it seems that the user is working with a custom embedding model in PyTorch. The model is trained using a custom loss function `MyMultipleNegativesRankingLoss`, which is a variant of the Multiple Negatives Ranking loss. The loss is calculated during the training process in the `train_model` function.

The loss value is already being printed during training if the `verbose` flag is set to `True`. This is done in the following line of code:

```python
if verbose:
    print_text(f"> [Epoch {epoch}] Current loss: {loss_value}\n", color="blue")
```

If the user wants to check the loss during fine-tuning, they can set the `verbose` flag to `True` when calling the `train_model` function. This will print the loss value at each training step.

The relevant files for this issue are:

1. `llama_index/finetuning/embeddings/adapter_utils.py`: This file contains the definition of the custom loss function and the training function for the model.

2. `llama_index/finetuning/types.py`: This file contains the abstract base classes for different types of fine-tuning engines. The user's Embedding model might be a subclass of one of these classes.
